internal/logger/target/http/http.go

```diff
@@ -97,70 +107,104 @@ func (h *Target) String() string {
+		return errors.New("target is closed")
 	}
 
-	req.Header.Set(xhttp.ContentType, "application/json")
+	// This will check if we can reach the remote.
+	checkAlive := func() error {
+		ctx, cancel := context.WithTimeout(context.Background(), 2*webhookCallTimeout)
+		defer cancel()
 
-	// Set user-agent to indicate MinIO release
-	// version to the configured log endpoint
-	req.Header.Set("User-Agent", h.config.UserAgent)
+		req, err := http.NewRequestWithContext(ctx, http.MethodPost, h.config.Endpoint, strings.NewReader(`{}`))
+		if err != nil {
+			return err
+		}
 
-	if h.config.AuthToken != "" {
-		req.Header.Set("Authorization", h.config.AuthToken)
-	}
+		req.Header.Set(xhttp.ContentType, "application/json")
 
-	// If proxy available, set the same
-	if h.config.Proxy != "" {
-		proxyURL, _ := url.Parse(h.config.Proxy)
-		transport := h.config.Transport
-		ctransport := transport.(*http.Transport).Clone()
-		ctransport.Proxy = http.ProxyURL(proxyURL)
-		h.config.Transport = ctransport
-	}
+		// Set user-agent to indicate MinIO release
+		// version to the configured log endpoint
+		req.Header.Set("User-Agent", h.config.UserAgent)
 
-	client := http.Client{Transport: h.config.Transport}
-	resp, err := client.Do(req)
-	if err != nil {
-		return err
-	}
-	h.client = &client
+		if h.config.AuthToken != "" {
+			req.Header.Set("Authorization", h.config.AuthToken)
+		}
 
-	// Drain any response.
-	xhttp.DrainBody(resp.Body)
+		resp, err := h.client.Do(req)
+		if err != nil {
+			return err
+		}
+		// Drain any response.
+		xhttp.DrainBody(resp.Body)
 
-	if !acceptedResponseStatusCode(resp.StatusCode) {
-		if resp.StatusCode == http.StatusForbidden {
-			return fmt.Errorf("%s returned '%s', please check if your auth token is correctly set",
+		if !acceptedResponseStatusCode(resp.StatusCode) {
+			if resp.StatusCode == http.StatusForbidden {
+				return fmt.Errorf("%s returned '%s', please check if your auth token is correctly set",
+					h.config.Endpoint, resp.Status)
+			}
+			return fmt.Errorf("%s returned '%s', please check your endpoint configuration",
 				h.config.Endpoint, resp.Status)
 		}
-		return fmt.Errorf("%s returned '%s', please check your endpoint configuration",
-			h.config.Endpoint, resp.Status)
+		return nil
 	}
 
-	h.lastStarted = time.Now()
-	h.online = true
-	atomic.AddInt64(&h.workers, 1)
-	go h.startHTTPLogger()
+	err = checkAlive()
+	if err != nil {
+		// Start a goroutine that will continue to check if we can reach
+		h.revive.Do(func() {
+			go func() {
+				t := time.NewTicker(time.Second)
+				defer t.Stop()
+				for range t.C {
+					if atomic.LoadInt32(&h.status) != statusOffline {
+						return
+					}
+					if err := checkAlive(); err == nil {
+						// We are online.
+						if atomic.CompareAndSwapInt32(&h.status, statusOffline, statusOnline) {
+							h.workerStartMu.Lock()
+							h.lastStarted = time.Now()
+							h.workerStartMu.Unlock()
+							atomic.AddInt64(&h.workers, 1)
+							go h.startHTTPLogger()
```

Comment:
why do we keep starting a new logger for every reconnect? what happens to the previous one? the problem seems to me that there is actually no control over startHTTPLogger(). 

Isn't it easier to simply just have X workers available always? like, have 8 workers always per server?  it should be sufficient to manage any high volume API calls.

Reply:
@harshavardhana I think you are misreading the code. This only spawns the first worker when we can reach the destination.

`h.revive.Do` only ever runs *once*. We return once we are able to spawn a worker.

It will scale up to 16 workers as needed, but that is not done here.